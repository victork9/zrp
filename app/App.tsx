import { StatusBar } from "expo-status-bar";
import { useState } from "react";
import { Alert, FlatList, Pressable, SafeAreaView, StyleSheet, Text, TextInput, View } from "react-native";
import { Root, getPockemons } from "./src/server/api";

export default function App() {
    const [data, setData] = useState<Root[] | []>([]);
    const [error, setError] = useState<boolean>(false);
    const [input, setInput] = useState("");
    const handleApi = async () => {
        setError(false);
        if (input) {
            try {
                const response: any = await getPockemons(input.trim().toLowerCase());
                if (response.data.status === 404) {
                    setError(true);
                } else {
                    setData(response.data);
                }
                return;
            } catch (error) {
                setError(true);
            }
        }
        Alert.alert("Atenção", "por favor preencha o campo");
    };
    return (
        <SafeAreaView style={styles.container}>
            <View testID="View_Container_Search" style={{ flexDirection: "row", width: "100%", justifyContent: "center", alignItems: "center" }}>
                <TextInput
                    value={input}
                    testID="TextInput_Search"
                    onChangeText={setInput}
                    style={styles.input}
                    placeholder="Please, inform your pockemon"
                />
                <Pressable testID="Pressable_Search" onPress={handleApi} style={styles.pressable}>
                    <Text style={styles.text}>Pesquisar</Text>
                </Pressable>
            </View>
            {error && <Text style={styles.textHeader}>Erro na consulta</Text>}
            {data?.length > 0 && (
                <FlatList
                testID="Flatlist_Container"
                    style={styles.flatlistContainer}
                    data={data}
                    ListHeaderComponent={<Text style={styles.textHeader}>Lista das habilidades</Text>}
                    renderItem={({ item, index }) => (
                        <View style={styles.viewList}>
                            <Text>Nome da habilidade:</Text>
                            <Text>{item.ability.name}</Text>
                        </View>
                    )}
                />
            )}

            <StatusBar style="auto" />
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        marginHorizontal: 16,
        marginTop: 50,
    },
    input: { width: "60%", padding: 8, height: 40, borderRadius: 4, borderWidth: 1 },
    pressable: { padding: 12, backgroundColor: "blue", width: "30%", alignItems: "center", borderRadius: 24, margin: 16 },
    text: { color: "white" },
    viewList: {
        flexDirection: "row",
        gap: 4,
        backgroundColor: "#dcdcdc",
        borderRadius: 8,
        marginTop: 8,
        paddingHorizontal: 8,
        paddingVertical: 12,
    },
    textHeader: { fontSize: 24 },
    flatlistContainer: { flex: 1 },
});
