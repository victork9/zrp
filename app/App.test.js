import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";

jest.mock("./src/server/api", () => ({
    getPockemons: jest.fn().mockResolvedValue({
        data: [
            {
                ability: {
                    name: "imposter",
                    url: "https://pokeapi.co/api/v2/ability/150/",
                },
                is_hidden: true,
                slot: 3,
            },
            {
                ability: {
                    name: "limber",
                    url: "https://pokeapi.co/api/v2/ability/7/",
                },
                is_hidden: false,
                slot: 1,
            },
        ],
        status: 200,
    }),
}));

import App from "./App";

describe("Testing app", () => {
    it("Should render corretly", () => {
        const tree = render(<App />);
        expect(tree).toMatchSnapshot();
    });
    it("Should find elements", () => {
        const tree = render(<App />);
        const findTexts = tree.getByTestId("View_Container_Search");
        expect(findTexts).toBeTruthy();
    });
    it("Should change TextInput", () => {
        const tree = render(<App />);
        const findTexts = tree.getByTestId("TextInput_Search");

        fireEvent.changeText(findTexts, "ditto");
        expect(findTexts.props.value).toBe("ditto");
    });
    it("Should call api success", () => {
        const tree = render(<App />);
        const findTexts = tree.getByTestId("TextInput_Search");
        const pressable = tree.getByTestId("Pressable_Search");

        fireEvent.changeText(findTexts, "ditto");
        fireEvent.press(pressable);
        waitFor(() => {
            expect(tree.getByTestId("Flatlist_Container")).toBeTruthy();
        });
				expect(tree).toMatchSnapshot();
    });
});
