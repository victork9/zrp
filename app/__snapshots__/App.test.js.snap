// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing app Should call api 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginHorizontal": 16,
      "marginTop": 50,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="View_Container_Search"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Please, inform your pockemon"
      style={
        {
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "padding": 8,
          "width": "60%",
        }
      }
      testID="TextInput_Search"
      value="ditto"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 24,
          "margin": 16,
          "padding": 12,
          "width": "30%",
        }
      }
      testID="Pressable_Search"
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        Pesquisar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Testing app Should render corretly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginHorizontal": 16,
      "marginTop": 50,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="View_Container_Search"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Please, inform your pockemon"
      style={
        {
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "padding": 8,
          "width": "60%",
        }
      }
      testID="TextInput_Search"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 24,
          "margin": 16,
          "padding": 12,
          "width": "30%",
        }
      }
      testID="Pressable_Search"
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        Pesquisar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
